siliconcompiler/remote/client.py

```diff
@@ -196,30 +196,116 @@ def remote_run(chip):
 
     # Check the job's progress periodically until it finishes.
     is_busy = True
+    all_nodes = []
+    result_procs = []
+    for step in chip.get('option', 'steplist'):
+        for index in chip.getkeys('flowgraph', chip.get('option', 'flow'), step):
+            all_nodes.append(f'{step}{index}')
+    completed = []
+    multiprocessor = multiprocessing.get_context('spawn')
     while is_busy:
         time.sleep(30)
         try:
             is_busy_info = is_job_busy(chip)
             is_busy = is_busy_info['busy']
             if is_busy:
-                if (':' in is_busy_info['message']):
-                    msg_lines = is_busy_info['message'].splitlines()
-                    cur_step = msg_lines[0][msg_lines[0].find(': ') + 2:]
-                    cur_log = '\n'.join(msg_lines[1:])
-                    chip.logger.info("Job is still running (%d seconds, step: %s)." % (
-                                     int(time.monotonic() - step_start), cur_step))
-                    if cur_log:
-                        chip.logger.info(f"Tail of current logfile:\n{cur_log}\n")
-                else:
-                    chip.logger.info("Job is still running (%d seconds, step: unknown)" % (
-                                     int(time.monotonic() - step_start)))
+                try:
```

Comment:
this block is quite long, could you please refactor.

Reply:
Yes - actually that's part of why I want to get this merged soon, to pull out the logic which prints `check_progress` results for the sake of the `sc-remote` CLI app. I guess I can just do it here.